agent/engine/docker_image_manager.go

```diff
@@ -379,70 +384,75 @@ func (imageManager *dockerImageManager) getNonECSContainerIDs(ctx context.Contex
 	return nonECSContainersIDs, nil
 }
 
-type ImageWithSize struct {
+type ImageWithSizeID struct {
 	ImageName string
+	ImageID   string
 	Size      int64
 }
 
 func (imageManager *dockerImageManager) removeNonECSImages(ctx context.Context, nonECSImagesNumToDelete int) {
 	if nonECSImagesNumToDelete == 0 {
 		return
 	}
-	var nonECSImageNames = imageManager.getNonECSImageNames(ctx)
-	var nonECSImageNamesRemoveEligible []string
-	for _, nonECSImage := range nonECSImageNames {
-		if !isInExclusionList(nonECSImage, imageManager.imageCleanupExclusionList) {
-			nonECSImageNamesRemoveEligible = append(nonECSImageNamesRemoveEligible, nonECSImage)
+	nonECSImages := imageManager.getNonECSImages(ctx)
+	var nonECSImagesRmEligible []ImageWithSizeID
+	for _, nonECSImage := range nonECSImages {
+		if !isInExclusionList(nonECSImage.ImageName, imageManager.imageCleanupExclusionList) {
+			nonECSImagesRmEligible = append(nonECSImagesRmEligible, nonECSImage)
 		}
 	}
 
-	var imageWithSizeList []ImageWithSize
-	for _, imageName := range nonECSImageNamesRemoveEligible {
-		resp, iiErr := imageManager.client.InspectImage(imageName)
-		if iiErr != nil {
-			seelog.Errorf("Error inspecting non-ECS image name: %s - %v", imageName, iiErr)
+	// Get the all image sizes
+	for _, image := range nonECSImagesRmEligible {
+		resp, err := imageManager.client.InspectImage(image.ImageID)
```

Comment:
any reason why this was change from imageName to imageID? 
Sometimes, two images can have same ID but different names. just making sure this doesn't break anything.

Reply:
The reason I changed this to image ID is because with image name we don't properly cleanup 'dangling' images. I made a comment about this here: https://github.com/aws/amazon-ecs-agent/issues/1684#issuecomment-488866388

But that is a good catch, I did not realize two images can have the same ID but a different name. Maybe changing this behavior should be broken out into a separate issue and PR, if we want to do it at all.

Reply:
okay, I've removed the parts that removed 'dangling images'